core/src/main/scala/org/apache/spark/scheduler/DAGScheduler.scala

```diff
@@ -1365,19 +1375,43 @@ class DAGScheduler(
    */
   private[scheduler] def handleExecutorLost(
       execId: String,
-      filesLost: Boolean,
-      maybeEpoch: Option[Long] = None) {
+      workerLost: Boolean): Unit = {
+    // if the cluster manager explicitly tells us that the entire worker was lost, then
+    // we know to unregister shuffle output.  (Note that "worker" specifically refers to the process
+    // from a Standalone cluster, where the shuffle service lives in the Worker.)
+    val filesLost = workerLost || !env.blockManager.externalShuffleServiceEnabled
+    removeExecutorAndUnregisterOutputs(
+      execId = execId,
+      fileLost = filesLost,
```

Comment:
The `fileLost` vs. `filesLost` naming difference is a little confusing -- is the distinction even conveying a difference worth paying attention to?